### ISSUE TITLE
Nodes and headline links

### ISSUE THREAD

**Issue by fkgruber** [org-roam/org-roam-server#63]
Hi
org-roam now supports headline links. This is particularly useful when we have whole books that we are reading with org-noter and want to point to specific sections. However, currently org-roam-server just points to the file.  It seems that it would be more useful if it creates a new node to that headline. And maybe automatically add a link between that headline node and the file node

thanks
FKG

**Comment by goktug97** [comment]
As far as I know the headlines are stored as unique ids with their corresponding files. So I think It is not possible to add them as meaningful nodes without parsing the corresponding files.

```elisp
    (headlines
     [(id :unique :primary-key)
      (file :not-null)])
```

**Comment by goktug97** [comment]
Do you have any updates?

**Comment by theotheo** [comment]
I think it's very handy functionality. So is it quite hard to parse files?

**Comment by goktug97** [comment]
It will cause drop in performance and the function runs every second. I think parsing every file that has headlines might cause performance issues. People already experiencing performance issues and because of that I added org-roam-server-network-poll variable.

Also having nodes which are named "Introduction" will clutter the graph. It can be grouped into one node but grouped nodes don't support some of the functions that are used.

**Comment by theotheo** [comment]
What if we use cache parsed files? Is it would help to keep performance at an acceptable level?

**Comment by goktug97** [comment]
I don't know, it feels a little bit hacky. Also there are still cluttering problem and lack of functionality in clusters. If someone wants to try to implement this, can send a PR.

**Comment by theotheo** [comment]
BTW, I think to implement clusters or something! Is there any recommendations? Maybe we should open the separate issue to discuss clustering?

**Comment by goktug97** [comment]
https://visjs.github.io/vis-network/docs/network/#methodClustering

It will probably require lots of changes in the current code.
